monkey/tests/unit_tests/infection_monkey/command_builders/test_linux_agent_command_builder.py

```diff
@@ -0,0 +1,134 @@
+from pathlib import PurePosixPath
+
+import pytest
+from agentpluginapi import (
+    DropperExecutionMode,
+    IAgentCommandBuilderFactory,
+    ILinuxAgentCommandBuilder,
+    LinuxDownloadMethod,
+    LinuxDownloadOptions,
+    LinuxRunOptions,
+)
+from monkeytypes import AgentID
+
+from infection_monkey.model import DROPPER_ARG, MONKEY_ARG
+
+AGENT_DESTINATION_PATH = PurePosixPath("/tmp/agent")
+DROPPER_DESTINATION_PATH = PurePosixPath("/tmp/dropper")
+EXPECTED_AGENT_DESTINATION_PATH = str(AGENT_DESTINATION_PATH)
+EXPECTED_DROPPER_DESTINATION_PATH = str(DROPPER_DESTINATION_PATH)
+DOWNLOAD_URL = "http://127.0.0.1/agent"
+
+
+@pytest.fixture
+def linux_agent_command_builder(
+    agent_command_builder_factory: IAgentCommandBuilderFactory,
+) -> ILinuxAgentCommandBuilder:
+    return agent_command_builder_factory.create_linux_agent_command_builder()
+
+
+def test_initial_command(linux_agent_command_builder: ILinuxAgentCommandBuilder):
+    assert linux_agent_command_builder.get_command() == ""
+
+
+@pytest.mark.parametrize(
+    "expected_method, not_expected_method, download_method",
+    [
+        ("wget", "curl", LinuxDownloadMethod.WGET),
+        ("curl", "wget", LinuxDownloadMethod.CURL),
+    ],
+)
+def test_build_download_command(
+    linux_agent_command_builder: ILinuxAgentCommandBuilder,
+    expected_method: str,
+    not_expected_method: str,
+    download_method: LinuxDownloadMethod,
+):
+    linux_download_options = LinuxDownloadOptions(
+        agent_destination_path=AGENT_DESTINATION_PATH,
+        download_method=download_method,
+        download_url=DOWNLOAD_URL,
+    )
+    linux_agent_command_builder.build_download_command(linux_download_options)
+
+    actual_command = linux_agent_command_builder.get_command()
+    assert expected_method in actual_command
+    assert not_expected_method not in actual_command
+    assert "chmod" in actual_command
+    assert EXPECTED_AGENT_DESTINATION_PATH in actual_command
```

Comment:
Tests should follow the Arrange, Act, Assert (AAA) pattern. As much as possible, formatting the test according to AAA makes it easier to read (see suggestion below). Please check the formatting on all tests.

```suggestion
def test_build_download_command(
    linux_agent_command_builder: ILinuxAgentCommandBuilder,
    expected_method: str,
    not_expected_method: str,
    download_method: LinuxDownloadMethod,
):
    linux_download_options = LinuxDownloadOptions(
        agent_destination_path=AGENT_DESTINATION_PATH,
        download_method=download_method,
        download_url=DOWNLOAD_URL,
    )
    
    linux_agent_command_builder.build_download_command(linux_download_options)
    actual_command = linux_agent_command_builder.get_command()
    
    assert expected_method in actual_command
    assert not_expected_method not in actual_command
    assert "chmod" in actual_command
    assert EXPECTED_AGENT_DESTINATION_PATH in actual_command
```